src/Psr/SimpleCache/SimpleCacheDecorator.php

```diff
@@ -42,22 +42,21 @@ class SimpleCacheDecorator implements SimpleCacheInterface
      */
     public const INVALID_KEY_CHARS = ':@{}()/\\';
 
-    /** @var bool */
-    private $providesPerItemTtl = true;
+    private bool $providesPerItemTtl = true;
 
-    /** @var StorageInterface */
-    private $storage;
+    private StorageInterface $storage;
 
     /**
      * Reference used by storage when calling getItem() to indicate status of
      * operation.
      *
      * @var null|bool
+     *
+     * Use ?bool type make it unserialized, even with null initialized
```

Comment:
This is confusing 🤔

Reply:
Yes, I don't know what possible values are /cc @boesing

Reply:
with `?bool $success = null`, it got error:

```bash
There were 11 errors:

1) LaminasTest\Cache\Psr\SimpleCache\SimpleCacheIntegrationTest::testSetTtl
Laminas\Cache\Psr\SimpleCache\SimpleCacheException: Serialized data must be a string containing serialized PHP code; received: boolean

/Users/samsonasik/www/laminas-cache/src/Psr/SimpleCache/SimpleCacheDecorator.php:322
/Users/samsonasik/www/laminas-cache/src/Psr/SimpleCache/SimpleCacheDecorator.php:91
/Users/samsonasik/www/laminas-cache/vendor/laminas/laminas-cache-storage-adapter-test/src/AbstractSimpleCacheIntegrationTest.php:220
```

Reply:
I am not sure if I understand what you try to outline with `unserialized`.
What I can say is that `laminas-cache` has mostly no typing and is passing plenty of stuff by-reference.
I do not really "trust" upstream as of now and thus, can't say if `success` is always boolean.

Therefore, I'd say lets keep it as is.
But I definitely do not get what the point of the comment is.

Reply:
@boesing ok, removed comment https://github.com/laminas/laminas-cache/pull/209/commits/79a244dda045207056b366df65fea41bc707f2d6 👍

Reply:
Can you please tell me what you wanted to point out with the comment? 
Because I just did not understood it.
Its not that I wanted to get rid of it, just wanted to have some more backstory 😅

Reply:
It has `@var null|bool` in the docblock, so it assume it should be `?bool`, and if it doesn't has initialization in `__construct`, it means it needs to have default `null` value, so the type should be:

```php
private ?bool $success = null;
```

Reply:
> with `?bool $success = null`, it got error:
> 
> ```shell
> There were 11 errors:
> 
> 1) LaminasTest\Cache\Psr\SimpleCache\SimpleCacheIntegrationTest::testSetTtl
> Laminas\Cache\Psr\SimpleCache\SimpleCacheException: Serialized data must be a string containing serialized PHP code; received: boolean
> 
> /Users/samsonasik/www/laminas-cache/src/Psr/SimpleCache/SimpleCacheDecorator.php:322
> /Users/samsonasik/www/laminas-cache/src/Psr/SimpleCache/SimpleCacheDecorator.php:91
> /Users/samsonasik/www/laminas-cache/vendor/laminas/laminas-cache-storage-adapter-test/src/AbstractSimpleCacheIntegrationTest.php:220
> ```


TBH: I have no idea what this error is about. Maybe its worth digging into it? But anyways, keeping the code as is might be suitable for now. I'd rather go without this one property optimized for PHP 7.4 than breaking upstream code because of this. 👍🏼